### PR TITLE
More java ATH

### PR DESCRIPTION
**Pull request by imeredith**
A few tests ported from JS. But more importantly support for Pipeline jobs which JS ath uses a lot. As well as some refactoring

### CODE REVIEW

File: acceptance-tests/src/main/java/io/blueocean/ath/model/MultiBranchPipeline.java [hunk 1]
```diff
@@ -12,7 +12,7 @@
 import java.io.IOException;
 import java.util.List;
 
-public class MultiBranchPipeline extends Pipeline{
+public class MultiBranchPipeline extends AbstractPipeline {
```
Comment (cliffmeyers) [inline acceptance-tests/src/main/java/io/blueocean/ath/model/MultiBranchPipeline.java]: Should this override `isMultibranch()` to return true?

File: acceptance-tests/src/main/java/io/blueocean/ath/pages/blue/RunDetailsArtifactsPage.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+    @Inject
+    public RunDetailsArtifactsPage(WebDriver driver, @Assisted AbstractPipeline pipeline) {
+        this.driver = driver;
+        this.pipeline = pipeline;
+        PageFactory.initElements(driver, this);
+    }
+
+    public void open(String pipeline, Integer runNumber) {
+        driver.get(base+"/blue/organizations/jenkins/"+ pipeline + "/detail/master/" + runNumber +"/artifacts");
```
Comment (cliffmeyers) [inline acceptance-tests/src/main/java/io/blueocean/ath/pages/blue/RunDetailsArtifactsPage.java]: `pipeline` probably needs URI encoding to support jobs nested in folders?

File: acceptance-tests/src/main/java/io/blueocean/ath/pages/blue/RunDetailsArtifactsPage.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    public void checkUrl(String branch, int runNumber) {
+        wait.until(ExpectedConditions.urlContains(getUrl(branch, runNumber)), 30000);
+    }
+    public String getUrl(int runNumber) {
+        return getUrl(null, runNumber);
+    }
+
+    public String getUrl(String branch, int runNumber) {
+        if(pipeline.isMultiBranch()) {
+            String tempBranch = branch == null ? "master" : branch;
```
Comment (cliffmeyers) [inline acceptance-tests/src/main/java/io/blueocean/ath/pages/blue/RunDetailsArtifactsPage.java]: May also need URI encoding for branch name.

File: acceptance-tests/src/main/java/io/blueocean/ath/model/Pipeline.java [hunk 4]
```diff
@@ -1,81 +1,44 @@
...
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.Files;
 
-    public Folder getFolder() {
-        return folder;
+public class Pipeline extends AbstractPipeline {
```
Comment (cliffmeyers) [inline acceptance-tests/src/main/java/io/blueocean/ath/model/Pipeline.java]: `ClassicPipeline` might be a better name, just to differentiate it from our more generic use of "pipeline" throughout.

File: acceptance-tests/src/main/java/io/blueocean/ath/model/MultiBranchPipeline.java (hunk 1 above)
Comment (imeredith) [reply acceptance-tests/src/main/java/io/blueocean/ath/model/MultiBranchPipeline.java]: yes

File: acceptance-tests/src/main/java/io/blueocean/ath/model/Pipeline.java (hunk 4 above)
Comment (imeredith) [reply acceptance-tests/src/main/java/io/blueocean/ath/model/Pipeline.java]: Ah yeah, i was trying to figure our job vs pipeline. Sounds lke a good idea

**Comment by michaelneale** [comment]
@imeredith you picking this up?

**Comment by michaelneale** [comment]
LGTM - if @cliffmeyers is satisfied with his objections above

**Pull request merged**